crates/common/consensus/src/deneb/beacon_state.rs

```diff
@@ -581,4 +582,37 @@ impl BeaconState {
         }
         Ok((rewards, penalties))
     }
+
+    pub fn process_block_header(mut self, block: BeaconBlock) -> anyhow::Result<()> {
+        ensure!(
+            self.slot == block.slot,
+            "State slot must be equal to block slot"
+        );
+        ensure!(
+            block.slot > self.latest_block_header.slot,
+            "Block slot must be greater than latest block header slot of state"
+        );
+        ensure!(
+            block.proposer_index == self.get_beacon_proposer_index().unwrap(),
+            "Block proposer index must be equal to beacon proposer index"
+        );
+        ensure!(
+            block.parent_root == self.latest_block_header.tree_hash_root(),
+            "Block Parent Root must be equal root of latest block header"
+        );
+
+        self.latest_block_header = BeaconBlockHeader {
```

Comment:
```suggestion
        // Cache current block as the new latest block
        self.latest_block_header = BeaconBlockHeader {
```

It would be nice to carry over the comments from the pseudo code